signbank/dictionary/models.py

```diff
@@ -599,6 +601,110 @@ class Meta:
     def __str__(self):
         return self.idgloss
 
+    @classmethod
+    def get_field_names(cls):
+        fields = cls._meta.get_fields(include_hidden=True)
+        return [field.name for field in fields if field.concrete]
+
+    @classmethod
+    def get_field(cls, field):
+        field = cls._meta.get_field(field)
+        return field
+
+    def get_wordclasses_display(self):
+        return ", ".join([d.english_name for d in self.wordclasses.all()])
+
+    def get_fields_dict(self, fieldnames):
+        """
+        This function is copied from Global Signbank.
+        It has been adapted to work for NZSL's data structure.
+        """
+
+        # TO DO include other gloss relations in the fieldnames (e.g., simultaneous morphology, below)
+        # a way to determine w/o hard-coding if a fieldname is for a related model
+        # then can use the above display methods
+        # related_models = [rel.related_model._meta.model_name for rel in self._meta.related_objects]
+        # print(related_models)
+        # requires changing search field names
+
+        fields = {}
+        site = Site.objects.get_current()
+        if 'idgloss' in fieldnames:
+            # this is the default setting API_FIELDS
+            fields['idgloss'] = self.idgloss
+
+        gloss_fields = [Gloss.get_field(fname) for fname in Gloss.get_field_names()]
+        fields_data = []
+        for field in gloss_fields:
+            if hasattr(field, 'field_choice_category'):
+                fc_category = field.field_choice_category
+            else:
+                fc_category = None
+            fields_data.append((field.name, field.verbose_name.title(), fc_category))
+
+        # gloss translations
+        if self.dataset:
+            for language in self.dataset.translation_languages.all():
+                glosstranslations = self.glosstranslations_set.filter(language=language)
+                if glosstranslations and glosstranslations.count() > 0:
```

Comment:
**nitpick**: `len(query) and bool(query)` are usually better than using `.count()` or `.exists()` as they don't trigger a new query